design/httpproxy-conflict.md

```diff
@@ -0,0 +1,82 @@
+# HTTPProxy Conflict Resolution Design
+
+Status: Draft
+
+## Abstract
+
+Configuration errors in HTTPProxy resources can result in objects being marked as invalid and conversely will not serve traffic.
+These errors can come from invalid route configuration, missing service references, or invalid includes to other resources. 
+
+## Background
+HTTPProxy is designed as a multi-team aware Ingress, reduce annotation sprawl by providing a sensible home for configuration items.
+Additionally, HTTPProxy offers a feedback loop to users via a `Status` field on the object.
+Contour has further extended this to add `Conditions` which allows for a set of `Errors` and `Warnings` which inform users of errors in the HTTPProxy resource.
+
+When the first error or warning is detected, these fields are updated and in the status of the HTTPProxy object, however if there are more errors, they are not all shown to the user, only the first.
+There are ongoing efforts to change this logic to inform of all the errors or warnings dealing with a specific object.
+
+Contour currently invalidates the entire resource if a configuration error is encountered which can cause downtime for specific routes. 
+
+## Goals
+- Serve valid traffic configurations if portions of the spec are invalid
+
+## Non Goals
+- Surface a complete set errors or warnings for a resource (this is a different issue)
```

Comment:
I think it's important to ensure that we surface which parts are invalid though.

Reply:
Maybe enough to mention a specific other issue URL here?

Reply:
> I think it's important to ensure that we surface which parts are invalid though.
Yup 100%, that won't go away.

The point of this statement was to say that this change isn't going to do what I think https://github.com/projectcontour/contour/issues/2495 was intending to do. 

Today we show a single error with HTTPProxy and bubble that back up. Conditions were added which allows for multiple errors/warnings to be set on the object. 

My original thought was that I didn't want to tackle that work in this design, but possibly it might need to since it will require a change to the objects are processed.